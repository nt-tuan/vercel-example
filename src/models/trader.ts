import { KYCStatus } from "./kyc";

export interface Trader {
  marketplace: string;
  username: string;
  email: string;
  dateJoined: string;
  status: KYCStatus;
}

export interface TraderContact {
  companyName?: string;
  phoneNumber?: string;
  email?: string;
  country?: string;
  address?: string;
  emailVerified: boolean;
}
