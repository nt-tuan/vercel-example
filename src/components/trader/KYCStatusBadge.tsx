import { Chip } from "@mui/material";
import { KYCStatus, KYCStatusLabel } from "../../models/kyc";

const statusColor: {
  [key in KYCStatus]: "error" | "warning" | "info" | "success" | "default";
} = {
  [KYCStatus.FAILED]: "error",
  [KYCStatus.IN_PROGRESS]: "warning",
  [KYCStatus.PENDING_EMAIL]: "info",
  [KYCStatus.PASSED]: "success",
  [KYCStatus.EMAIL_VERIFIED]: "success",
  [KYCStatus.INACTIVE]: "default",
};
interface Props {
  status?: KYCStatus;
  size?: "small" | "medium";
}
export const KYCStatusBadge = ({ status, size }: Props) => {
  if (!status) return <></>;
  return (
    <Chip
      color={statusColor[status]}
      label={KYCStatusLabel[status] ?? "default"}
      size={size}
    />
  );
};
