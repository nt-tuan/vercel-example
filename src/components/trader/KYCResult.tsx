import React from "react";
import { KYCResult as Model, KYCStatus } from "models/kyc";
import { DataGrid, GridColDef, GridValueGetterParams } from "@mui/x-data-grid";
import { IconButton } from "@material-ui/core";
import VisibilityIcon from "@mui/icons-material/Visibility";
import { KYCStatusBadge } from "./KYCStatusBadge";
import { Dialog, DialogContent, DialogTitle } from "@mui/material";
interface Props {
  data?: Model[];
}

export const KYCResult = ({ data }: Props) => {
  const [message, setMessage] = React.useState<string>();
  const handleRowClick = () => {};
  const viewMessage = (message?: string) => {
    setMessage(message);
  };
  const columns: GridColDef[] = [
    {
      field: "id",
      hide: true,
    },
    {
      field: "date",
      headerName: "Date",
    },
    {
      field: "type",
      headerName: "Type",
    },
    {
      field: "provider",
      headerName: " Provider",
      flex: 1,
    },
    {
      field: "status",
      headerName: "Status",
      width: 130,
      renderCell: (params: GridValueGetterParams) => (
        <KYCStatusBadge status={params.value as KYCStatus} />
      ),
    },
    {
      field: "action",
      headerName: "Action",
      renderCell: (params: GridValueGetterParams) => {
        return (
          <IconButton onClick={() => viewMessage(params.row.message)}>
            <VisibilityIcon />
          </IconButton>
        );
      },
    },
  ];
  return (
    <>
      <DataGrid
        rows={data ?? []}
        columns={columns}
        loading={data == null}
        hideFooter
        disableColumnMenu
        autoHeight
        onRowClick={handleRowClick}
      />
      <Dialog
        onClose={() => setMessage(undefined)}
        open={message != null}
        fullWidth
      >
        <DialogTitle>KYC request and response </DialogTitle>
        <DialogContent>{message}</DialogContent>
      </Dialog>
    </>
  );
};
