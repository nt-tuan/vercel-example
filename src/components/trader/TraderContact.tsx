import {
  Card,
  CardContent,
  List,
  ListItem,
  Skeleton,
  Typography,
} from "@mui/material";
import React from "react";
import { TraderContact as Model } from "models/trader";
import { styled } from "@mui/system";
import { KYCStatusBadge } from "./KYCStatusBadge";
import { KYCStatus } from "models/kyc";

const Flex = styled("div")({
  width: "100%",
  display: "flex",
  justifyContent: "space-between",
});
const FlexItem = styled("div")({
  flex: "1 1 0",
  padding: "16px 0 0 0",
});

const WrappedItem = ({
  label,
  value,
}: {
  label: React.ReactNode;
  value: React.ReactNode;
}) => {
  return (
    <ListItem disablePadding>
      <Flex>
        <FlexItem>{label}</FlexItem>
        <FlexItem>{value}</FlexItem>
      </Flex>
    </ListItem>
  );
};

interface Props {
  contact?: Model;
  status?: KYCStatus;
}

export const TraderContact = ({ contact }: Props) => {
  if (contact == null)
    return <Skeleton variant="rectangular" width={500} height={300} />;
  return (
    <Card style={{ maxWidth: "500px" }}>
      <CardContent>
        <Typography fontWeight="bold">Contact Details</Typography>
        <List>
          <WrappedItem value={contact.companyName ?? ""} label="Company Name" />
          <WrappedItem value={contact.phoneNumber ?? ""} label="Phone Number" />
          <WrappedItem
            value={
              <div>
                <div>{contact.email}</div>
                {contact.emailVerified && (
                  <KYCStatusBadge
                    status={KYCStatus.EMAIL_VERIFIED}
                    size="small"
                  />
                )}
              </div>
            }
            label="Email Address"
          />
          <WrappedItem value={contact.country ?? ""} label="Country" />
          <WrappedItem value={contact.address ?? ""} label="Full Address" />
        </List>
      </CardContent>
    </Card>
  );
};
