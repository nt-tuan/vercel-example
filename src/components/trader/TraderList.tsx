import { Trader, TraderStatus } from "models/trader";
import {
  DataGrid,
  GridColDef,
  GridRowParams,
  GridValueGetterParams,
} from "@mui/x-data-grid";
import { TraderStatusBadge } from "./TraderStatusBadge";
import { useHistory } from "react-router-dom";

const columns: GridColDef[] = [
  {
    field: "marketplace",
    headerName: "Marketplace",
    width: 130,
    sortable: false,
    filterable: false,
    resizable: false,
  },
  {
    field: "username",
    headerName: "Username",
    minWidth: 130,
    flex: 1,
    sortable: false,
    filterable: false,
    resizable: false,
  },
  {
    field: "email",
    headerName: "Email",
    width: 130,
    sortable: false,
    filterable: false,
    resizable: false,
  },
  {
    field: "dateJoined",
    headerName: "Date Joined",
    sortable: false,
    filterable: false,
    resizable: false,
    width: 130,
  },
  {
    field: "status",
    headerName: "Status",
    width: 180,
    sortable: false,
    filterable: false,
    resizable: false,
    renderCell: (params: GridValueGetterParams) =>
      params.value && (
        <TraderStatusBadge
          status={
            TraderStatus[params.value.toString() as keyof typeof TraderStatus]
          }
        />
      ),
  },
];

export const TraderList = ({ traders }: { traders?: Trader[] }) => {
  const history = useHistory();
  const handleRowClick = (params: GridRowParams) => {
    history.push({ pathname: "/trader/" + params.row.id });
  };
  return (
    <DataGrid
      rows={traders ?? []}
      columns={columns}
      loading={traders == null}
      hideFooter
      disableColumnMenu
      autoHeight
      onRowClick={handleRowClick}
    />
  );
};
