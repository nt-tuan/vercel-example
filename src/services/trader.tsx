import { Trader, TraderContact } from "models/trader";
import { HttpAPI } from "./httpAPI";

const traderAPI = new HttpAPI("/core-services/dc-support");
export const getTraders: () => Promise<Trader[]> = async () => {
  return traderAPI.get<Trader[]>("/traders.json");
};

export const getTrader = async (id: string) => {
  return traderAPI.get<TraderContact>("/trader.json");
};
