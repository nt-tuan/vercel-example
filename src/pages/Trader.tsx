import React from "react";
import { TraderContact } from "components/trader/TraderContact";
import { TraderContact as Model, Trader as TraderModel } from "models/trader";
import { TraderVerifyingResult } from "models/trader";
import { useParams } from "react-router";
import {
  getTraderContact,
  getTraderVerifyingResult,
  getTrader,
} from "services/trader";
import { TraderVerifyingTable } from "components/trader/TraderVerifyingTable";
import { AdminLayout } from "layouts/AdminLayout";
import { Box, Skeleton, Typography } from "@mui/material";
import { WrappedLink } from "components/common/WrappedLink/WrappedLink";
interface PageParams {
  id: string;
}

const Trader = () => {
  const [contact, setContact] = React.useState<Model>();
  const [kycResult, setKYCResult] = React.useState<TraderVerifyingResult[]>();
  const [trader, setTrader] = React.useState<TraderModel>();
  const { id } = useParams<PageParams>();
  React.useEffect(() => {
    getTrader(id).then(setTrader);
    getTraderContact(id).then(setContact);
    getTraderVerifyingResult(id).then(setKYCResult);
  }, [id]);
  return (
    <>
      <AdminLayout.Header>
        {trader?.marketplace ?? <Skeleton variant="text" />} |{" "}
        {contact?.companyName ?? <Skeleton variant="text" />}
      </AdminLayout.Header>
      <AdminLayout.Breadcrumbs aria-label="breadcrumb">
        <WrappedLink color="inherit" underline="hover" href="/">
          Trader
        </WrappedLink>
        <Typography fontWeight="light">Trader Details</Typography>
      </AdminLayout.Breadcrumbs>
      <AdminLayout.Content>
        <TraderContact contact={contact} />
        <Box sx={{ marginTop: "24px" }}>
          <TraderVerifyingTable data={kycResult} />
        </Box>
      </AdminLayout.Content>
    </>
  );
};

export default Trader;
