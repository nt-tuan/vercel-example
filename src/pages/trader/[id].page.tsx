import React from "react";
import { TraderContact } from "components/trader/TraderContact";
import { TraderContact as Model } from "models/trader";
import { getTrader } from "services/trader";
import { TraderVerifyingTable } from "components/trader/TraderVerifyingTable";
import { AdminLayout, getLayout } from "layouts/AdminLayout";
import { Box, Skeleton, Typography } from "@mui/material";
import { WrappedLink } from "components/common/WrappedLink/WrappedLink";
import { useRouter } from "next/router";

const Page = () => {
  const [trader, setTrader] = React.useState<Model>();
  const router = useRouter();
  const { id } = router.query;

  React.useEffect(() => {
    if (typeof id !== "string") return;
    getTrader(id).then(setTrader);
  }, [id]);
  return (
    <>
      <AdminLayout.Header>
        {trader?.marketplace ?? <Skeleton variant="text" />} |{" "}
        {trader?.companyName ?? <Skeleton variant="text" />}
      </AdminLayout.Header>
      <AdminLayout.Breadcrumbs aria-label="breadcrumb">
        <WrappedLink
          color="inherit"
          underline="hover"
          href="/"
          fontWeight="bold"
        >
          Trader
        </WrappedLink>
        <Typography fontWeight="light">Trader Details</Typography>
      </AdminLayout.Breadcrumbs>
      <TraderContact contact={trader} />
      <Box sx={{ marginTop: "24px" }}>
        <TraderVerifyingTable data={trader?.verifyingResult} />
      </Box>
    </>
  );
};

Page.getLayout = getLayout;
export default Page;
