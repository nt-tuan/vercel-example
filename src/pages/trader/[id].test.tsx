import { render, act } from "@testing-library/react";
import React from "react";
import Trader from "./[id].page";
import { TraderStatus } from "models/trader";
import * as traderService from "services/trader";
import * as mockRouter from "next/router";

jest.mock("services/trader");
jest.mock("next/router");
jest.mock("components/common/WrappedLink/WrappedLink", () => ({
  WrappedLink: () => <a />,
}));
const mockService = () => {
  jest
    .spyOn(traderService, "getTraderContact")
    .mockImplementation(async () => ({
      companyName: "AAA",
      phoneNumber: "12312451",
      email: "email@asd.cos",
      country: "Vietnam",
      address: "232 Akd ale",
      emailVerified: true,
    }));
  jest
    .spyOn(traderService, "getTraderVerifyingResult")
    .mockImplementation(async () => [
      {
        id: "1",
        date: "2021-1-1-2",
        type: "string",
        provider: "Raputel",
        status: TraderStatus.PASSED,
        message: "My message",
      },
    ]);
  jest.spyOn(traderService, "getTrader").mockImplementation(async () => ({
    id: "1",
    marketplace: "Hotel Supply B2B",
    username: "tuan",
    email: "tuan@tuan.tuan  ",
    dateJoined: "2021-10-11",
    status: TraderStatus.PASSED,
  }));

  const mockUseParams = mockRouter.useRouter as jest.Mock;
  mockUseParams.mockImplementation(() => ({ query: { id: "1" } }));
};
test("Trader should render correctly", async () => {
  mockService();
  let containerElement;
  await act(async () => {
    const { container } = render(<Trader />);
    containerElement = container;
  });
  expect(containerElement).toMatchSnapshot();
});
