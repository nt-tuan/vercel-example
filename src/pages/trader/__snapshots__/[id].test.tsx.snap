// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trader should render correctly 1`] = `
<div>
  <h2
    class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
  >
    Hotel Supply B2B
     |
     
    AAA
  </h2>
  <div
    class="MuiBox-root css-r4f27e"
  >
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a />
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
        >
          /
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-b9dwom-MuiTypography-root"
          >
            Trader Details
          </p>
        </li>
      </ol>
    </nav>
  </div>
  <div
    class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1do9ab6-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
      >
        Contact Details
      </p>
      <ul
        class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
        >
          <div
            class="css-1v6lj4c"
          >
            <div
              class="css-1dmwlu3"
            >
              Company Name
            </div>
            <div
              class="css-1dmwlu3"
            >
              AAA
            </div>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
        >
          <div
            class="css-1v6lj4c"
          >
            <div
              class="css-1dmwlu3"
            >
              Phone Number
            </div>
            <div
              class="css-1dmwlu3"
            >
              12312451
            </div>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
        >
          <div
            class="css-1v6lj4c"
          >
            <div
              class="css-1dmwlu3"
            >
              Email Address
            </div>
            <div
              class="css-1dmwlu3"
            >
              <div>
                <div>
                  email@asd.cos
                </div>
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorSuccess MuiChip-filledSuccess css-1srradg-MuiChip-root"
                >
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                  >
                    Email Verified
                  </span>
                </div>
              </div>
            </div>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
        >
          <div
            class="css-1v6lj4c"
          >
            <div
              class="css-1dmwlu3"
            >
              Country
            </div>
            <div
              class="css-1dmwlu3"
            >
              Vietnam
            </div>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
        >
          <div
            class="css-1v6lj4c"
          >
            <div
              class="css-1dmwlu3"
            >
              Full Address
            </div>
            <div
              class="css-1dmwlu3"
            >
              232 Akd ale
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="MuiBox-root css-178yklu"
  >
    <div
      aria-colcount="5"
      aria-multiselectable="false"
      aria-rowcount="1"
      class="MuiDataGrid-root MuiDataGrid-root MuiDataGrid-autoHeight"
      role="grid"
    >
      <div>
        <div />
      </div>
      <div
        class="MuiDataGrid-main"
      >
        <div
          class="MuiDataGrid-columnsContainer"
          style="min-height: 56px; max-height: 56px; line-height: 56px;"
        >
          <div
            aria-rowindex="1"
            class="MuiDataGrid-columnHeaderWrapper scroll"
            role="row"
            style="transform: translate3d(0px, 0px, 0px); min-width: 610px;"
          >
            <div
              aria-colindex="1"
              class="MuiDataGrid-columnHeader"
              data-field="date"
              role="columnheader"
              style="width: 100px; min-width: 100px; max-width: 100px;"
              tabindex="0"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                >
                  <div
                    aria-label=""
                    class="MuiDataGrid-columnHeaderTitle"
                    data-mui-internal-clone-element="true"
                  >
                    Date
                  </div>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator"
                style="min-height: 56px; opacity: 1;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 19V5h2v14z"
                  />
                </svg>
              </div>
            </div>
            <div
              aria-colindex="2"
              class="MuiDataGrid-columnHeader"
              data-field="type"
              role="columnheader"
              style="width: 100px; min-width: 100px; max-width: 100px;"
              tabindex="-1"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                >
                  <div
                    aria-label=""
                    class="MuiDataGrid-columnHeaderTitle"
                    data-mui-internal-clone-element="true"
                  >
                    Type
                  </div>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator"
                style="min-height: 56px; opacity: 1;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 19V5h2v14z"
                  />
                </svg>
              </div>
            </div>
            <div
              aria-colindex="3"
              class="MuiDataGrid-columnHeader"
              data-field="provider"
              role="columnheader"
              style="width: 130px; min-width: 130px; max-width: 130px;"
              tabindex="-1"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                >
                  <div
                    aria-label=""
                    class="MuiDataGrid-columnHeaderTitle"
                    data-mui-internal-clone-element="true"
                  >
                     Provider
                  </div>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator"
                style="min-height: 56px; opacity: 1;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 19V5h2v14z"
                  />
                </svg>
              </div>
            </div>
            <div
              aria-colindex="4"
              class="MuiDataGrid-columnHeader"
              data-field="status"
              role="columnheader"
              style="width: 180px; min-width: 180px; max-width: 180px;"
              tabindex="-1"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                >
                  <div
                    aria-label=""
                    class="MuiDataGrid-columnHeaderTitle"
                    data-mui-internal-clone-element="true"
                  >
                    Status
                  </div>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator"
                style="min-height: 56px; opacity: 1;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 19V5h2v14z"
                  />
                </svg>
              </div>
            </div>
            <div
              aria-colindex="5"
              class="MuiDataGrid-columnHeader"
              data-field="action"
              role="columnheader"
              style="width: 100px; min-width: 100px; max-width: 100px;"
              tabindex="-1"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                >
                  <div
                    aria-label=""
                    class="MuiDataGrid-columnHeaderTitle"
                    data-mui-internal-clone-element="true"
                  >
                    Action
                  </div>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator"
                style="min-height: 56px; opacity: 1;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 19V5h2v14z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          style="overflow: visible; width: 0px;"
        >
          <div
            class="MuiDataGrid-windowContainer"
            style="width: 0px; height: 108px;"
          >
            <div
              class="MuiDataGrid-window"
              style="top: 56px; overflow-y: hidden;"
            >
              <div
                class="MuiDataGrid-dataContainer"
                style="min-width: 610px; min-height: 52px;"
              >
                <div
                  class="MuiDataGrid-viewport"
                  style="min-width: 0; max-width: 0; max-height: 52px;"
                >
                  <div
                    class="MuiDataGrid-renderingZone"
                    style="max-height: 52px; width: 610px; transform: translate3d(0px, 0px, 0px);"
                  >
                    <div
                      aria-rowindex="2"
                      aria-selected="false"
                      class="MuiDataGrid-row"
                      data-id="1"
                      data-rowindex="0"
                      role="row"
                      style="max-height: 52px; min-height: 52px;"
                    >
                      <div
                        aria-colindex="1"
                        class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                        data-colindex="0"
                        data-field="date"
                        role="cell"
                        style="min-width: 100px; max-width: 100px; min-height: 52px; max-height: 52px; line-height: 51px;"
                        tabindex="-1"
                      >
                        2021-1-1-2
                      </div>
                      <div
                        aria-colindex="2"
                        class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                        data-colindex="1"
                        data-field="type"
                        role="cell"
                        style="min-width: 100px; max-width: 100px; min-height: 52px; max-height: 52px; line-height: 51px;"
                        tabindex="-1"
                      >
                        string
                      </div>
                      <div
                        aria-colindex="3"
                        class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                        data-colindex="2"
                        data-field="provider"
                        role="cell"
                        style="min-width: 130px; max-width: 130px; min-height: 52px; max-height: 52px; line-height: 51px;"
                        tabindex="-1"
                      >
                        Raputel
                      </div>
                      <div
                        aria-colindex="4"
                        class="MuiDataGrid-cell--withRenderer MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                        data-colindex="3"
                        data-field="status"
                        role="cell"
                        style="min-width: 180px; max-width: 180px; min-height: 52px; max-height: 52px; line-height: 51px;"
                        tabindex="-1"
                      >
                        <div
                          class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorSuccess MuiChip-filledSuccess css-1abfbpn-MuiChip-root"
                        >
                          <span
                            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                          >
                            KYC Passed
                          </span>
                        </div>
                      </div>
                      <div
                        aria-colindex="5"
                        class="MuiDataGrid-cell--withRenderer MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                        data-colindex="4"
                        data-field="action"
                        role="cell"
                        style="min-width: 100px; max-width: 100px; min-height: 52px; max-height: 52px; line-height: 51px;"
                        tabindex="-1"
                      >
                        <button
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                          tabindex="0"
                          title="View message"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="VisibilityIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="Mui-resizeTriggers"
        >
          <div
            class="expand-trigger"
          >
            <div
              style="width: 1px; height: 1px;"
            />
          </div>
          <div
            class="contract-trigger"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
